themes/default/content/blog/redis-cloud-provider/index.md

```diff
@@ -0,0 +1,305 @@
+date: 2023-09-10
+meta_desc: Learn how to use Pulumi to provision fully managed Redis Enterprise Cloud clusters and connect them to your AWS workloads.
+meta_image: redis-enterprise-cloud.png
+authors:
+    - josh-kodroff
+tags:
+    - redis-cloud
+    - aws
+---
+
+Redis Enterprise Cloud makes apps faster by providing an in-memory database that enables teams to create a real-time data platform. In this post, you'll learn how to enable connectivity between Redis Enterprise Cloud and your workloads using Pulumi and the Redis Cloud provider.
+
+<!--more-->
+
+{{% notes type="info" %}}
+The full code for this example is available at <https://github.com/pulumi/examples/tree/master/rediscloud-aws-ts>. The full code differs slightly from the example below in that it has some parameterization that makes the code more portable, whereas in this post we show hardcoded values for simplicity.
+{{% /notes %}}
+
+## About Redis Enterprise Cloud
+
+Redis Enterprise Cloud is a database as-a-service (DBaaS) that provides managed Redis clusters. Redis Enterprise Cloud clusters can be accessed via network peering in AWS or Google Cloud, or over the internet. The [Redis Cloud provider](https://www.pulumi.com/registry/packages/rediscloud/) allows you to manage Redis Enterprise Cloud resources in Pulumi programs.
+
+## Architecture
+
+The architecture you'll build will include:
+
+* A Redis Enterprise Cloud database cluster.
+* An AWS VPC across 3 Availability Zones, with public and private subnets.
+* VPC Peering to enable connectivity between the Redis cluster and the VPC.
+
+You'll also prove the architecture works by spinning up an EC2 instance and running simple commands that target the Redis cluster.
+
+The following diagram shows the major components of the architecture:
+
+![Architecture diagram of an AWS VPC across 3 availability zones, connected to Redis Enterprise Cloud via VPC peering](architecture.png)
+
+## First Steps, Redis Enterprise Cloud Resources and Peering
+
+{{% notes type="info" %}}
+Before you begin this tutorial, you will need to add a payment method in the [Redis Cloud console](https://app.redislabs.com/).
+{{% /notes %}}
+
+To begin, initialize your Pulumi program in an empty directory:
+
+```bash
+pulumi new typescript
+```
+
+Then, add a reference to the Redis Cloud provider:
+
+```bash
+npm i @rediscloud/pulumi-rediscloud
+```
+
+Then, in your `index.ts` file, add use the `getPaymentMethodOutput` function to obtain your payment information, and use the resulting card ID to create a Redis Subscription:
+
+{{% notes type="info" %}}
+The Redis Cloud provider does not support Fixed subscriptions. You'll need to create a Flexible subscription in order to use the provider to manage resources with Pulumi.
+{{% /notes %}}
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as rediscloud from "@rediscloud/pulumi-rediscloud";
+
+const card = rediscloud.getPaymentMethodOutput({
```

Comment:
Would it be better/is it possible to store these values as configuration values/secrets? I know it's only the last four digits of a credit card, but I still hate seeing that information in a program.

Reply:
This is intentional. The corresponding example in Pulumi/examples has it parameterized, but for easier reading I decided to hardcode it here. I explain in a note that there are subtle differences between the example and post code as well.